internal/pkg/cli/pipeline_init.go

```diff
@@ -137,43 +144,49 @@ func newInitPipelineOpts(vars initPipelineVars) (*initPipelineOpts, error) {
 		sel:              selector.NewSelect(prompter, ssmStore),
 		runner:           exec.NewCmd(),
 		fs:               &afero.Afero{Fs: afero.NewOsFs()},
+		wsAppName:        wsAppName,
 	}, nil
 }
 
-// Validate returns an error if the flag values passed by the user are invalid.
+// Validate returns an error if the optional flag values passed by the user are invalid.
 func (o *initPipelineOpts) Validate() error {
-	if o.appName == "" {
+	// This command must be executed in the app's workspace because the pipeline manifest and buildspec will be created and stored.
+	if o.wsAppName == "" {
 		return errNoAppInWorkspace
 	}
-	if _, err := o.store.GetApplication(o.appName); err != nil {
-		return err
+	if o.appName != "" && o.appName != o.wsAppName {
+			return fmt.Errorf("cannot specify app %s because the workspace is already registered with app %s", o.appName, o.wsAppName)
+		}
+	// Validate the app name.
+	if _, err := o.store.GetApplication(o.wsAppName); err != nil {
+		return fmt.Errorf("get application %s configuration: %w", o.wsAppName, err)
 	}
+	return nil
+}
 
+// Ask prompts for required fields that are not passed in and validates them.
+func (o *initPipelineOpts) Ask() error {
 	if o.repoURL != "" {
 		if err := o.validateURL(o.repoURL); err != nil {
 			return err
 		}
+	} else {
+		if err := o.askRepository(); err != nil {
+			return err
+		}
 	}
 
-	if o.environments != nil {
+	if len(o.environments) > 0 {
```

Comment:
q: previously, `askEnvs()` was always called and therefore `o.envConfigs` would be set. Now, if `o.environments` has values in it, `askEnvs()` _isn't_ called, so `o.envConfigs` won't be set. Is that what we want to happen?